### ISSUE TITLE
Pressable - onPressIn and onPressOut don't work with android screen reader

### ISSUE THREAD

**Issue by EJJSharpe** [facebook/react-native#33563]
### Description

When using the Pressable component, the onPressIn and onPressOut functions do not work when using a screen reader on android. The functionality works fine on iOS but when testing on a Samsung S10 using samsungs 'TalkBack' screen reader, I am unable to select anything. 

I would expect that double tapping the screen with the Pressable selected would cause these functions to trigger but they simply do nothing.

### Version

0.68

### Output of `npx react-native info`

System:
    OS: macOS 12.2.1
    CPU: (8) arm64 Apple M1
    Memory: 81.06 MB / 8.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.14.0 - /usr/local/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 8.3.1 - /usr/local/bin/npm
    Watchman: 2022.02.21.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.11.2 - /opt/homebrew/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.2, iOS 15.2, macOS 12.1, tvOS 15.2, watchOS 8.3
    Android SDK:
      API Levels: 29, 30
      Build Tools: 29.0.2, 29.0.3, 30.0.2, 30.0.3, 31.0.0
      System Images: android-30 | Google APIs ARM 64 v8a, android-30 | Google APIs Intel x86 Atom, android-31 | Google APIs ARM 64 v8a, android-31 | Google Play ARM 64 v8a
      Android NDK: Not Found
  IDEs:
    Android Studio: 2020.3 AI-203.7717.56.2031.7784292
    Xcode: 13.2.1/13C100 - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_282 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: Not Found
    react-native: Not Found
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

1. Open snack on expo go using an android device (I tested with Samsung S10)
2. Turn on screen reader (this is called TalkBack on samsung it may be called something else on other android phones)
3. Tap a star to select it
4. Double tap the screen

### Snack, code example, screenshot, or link to a repository

https://snack.expo.dev/9i5crhwMcC

**Comment by andresleecom** [comment]
Same here

**Comment by AdamGerthel** [comment]
Same here. I can't understand how this can pass into release builds! Would never have happened if it wasn't tied to a11y.

**Comment by blavalla** [comment]
This is actually a problem with the way that Android and Talkback handle "clicks" from accessibility services.

When using an accessibility service that uses "explore by touch" (the focus-cursor based navigation used by Talkback), Android handles all of the touch interactions in the TouchExplorer class (https://cs.android.com/android/_/android/platform/frameworks/base/+/f55958fcfd3ce855455f38aed07325ebae2b2906:services/accessibility/java/com/android/server/accessibility/gestures/TouchExplorer.java;l=84)

When a user triggers a doubleTap gesture, this class will activate the onClick handler for the focused view, and only if the accessibility service has the flag `FLAG_SEND_MOTION_EVENTS` enabled, will it send motion events such as ACTION_DOWN and ACTION_UP to the service. 

However, Talkback does not currently have the `FLAG_SEND_MOTION_EVENTS` flag enabled. 

They seem to be experimenting with this as a feature, as they do have it enabled here (https://github.com/google/talkback/blob/771de7cdbf55b6adb4ca4c64c27a52584f2337cc/talkback/src/main/java/com/google/android/accessibility/talkback/TalkBackService.java#L1832)

But that is behind a check for `GestureReporter.ENABLED`, which always returns false.

So the short answer is that this is something that Talkback will need to add support for first, and once that happens React Native might automatically work, or we might need to adjust how our onPressIn and onPressOut events are triggered. But for the time being, there's unfortunately nothing we can do here.
